core/node/api_server/src/web3/namespaces/unstable.rs

```diff
@@ -55,8 +55,15 @@ impl UnstableNamespace {
                 signature: proof.signature,
                 proof: proof.proof,
                 proved_at: DateTime::<Utc>::from_naive_utc_and_offset(proof.updated_at, Utc),
+                status: proof.status,
                 attestation: proof.attestation,
             })
-            .collect::<Vec<_>>())
+            .collect::<Vec<_>>();
+
+        if proofs.is_empty() {
+            return Err(Web3Error::NoBlock);
+        }
```

Comment:
This doesn't look like a correct error to return in this case; the request queries by an L1 batch number, not a block. Is returning an empty array unacceptable? AFAIU, Web3 methods generally prefer to return `null` / "default" values instead of errors (e.g., when querying non-existing blocks / transactions).

Reply:
I [reverted it](https://github.com/matter-labs/zksync-era/pull/3321/files#diff-304621acb62844c4feb56448d7603d9401081bce25773e8f150f40a31ac5263c). ✅